tests/restapi/multiple_ssdindex_server.cpp

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+#include <ctime>
+#include <functional>
+#include <iomanip>
+#include <string>
+#include <cstdlib>
+#include <codecvt>
+
+#include <restapi/server.h>
+
+using namespace diskann;
+
+std::unique_ptr<Server>                           g_httpServer(nullptr);
+std::vector<std::unique_ptr<diskann::BaseSearch>> g_ssdSearch;
+
+void setup(const utility::string_t& address, const std::string& typestring) {
+  web::http::uri_builder uriBldr(address);
+  auto                   uri = uriBldr.to_uri();
+
+  std::cout << "Attempting to start server on " << uri.to_string() << std::endl;
+
+  g_httpServer =
+      std::unique_ptr<Server>(new Server(uri, g_ssdSearch, typestring));
+  std::cout << "Created a server object" << std::endl;
+
+  g_httpServer->open().wait();
+  ucout << U"Listening for requests on: " << address << std::endl;
+}
+
+void teardown(const utility::string_t& address) {
+  g_httpServer->close().wait();
+}
+
+int main(int argc, char* argv[]) {
+  if (argc != 6 && argc != 7) {
+    std::cout << "Usage: server ip_addr:port <float/int8/uint8> "
+                 "index_prefix_paths num_nodes_to_cache num_threads <tags_file>"
```

Comment:
Nitpick: the usage string doesn't follow normal conventions of mandatory and optional parameters.

Reply:
Maybe instead of num_threads, we could take "L num_threads BW" as a single string?